tsl/src/nodes/gapfill/planner.c

```diff
@@ -573,3 +575,96 @@ gapfill_adjust_window_targetlist(PlannerInfo *root, RelOptInfo *input_rel, RelOp
 		}
 	}
 }
+
+/*
+ * Check if it is safe to push down gapfill to data nodes.
+ * Currently, we allow only in the following cases,
+ *
+ * 1. when only one data node has all the chunks
+ * 2. when relation has at least one closed dimension and chunks
+ *    do not overlap across data nodes.
+ * 3. when group by matches space dimension
+ *    and is not an expression of space dimension.
+ */
+bool
+pushdown_gapfill(PlannerInfo *root, RelOptInfo *hyper_rel, Hyperspace *hs,
+				 DataNodeChunkAssignments *scas)
+{
+	const Dimension *dim;
+	ListCell *lc;
+	TargetEntry *tle;
+	bool space_dim_in_group_by = false;
+
+	Query *parse = root->parse;
+	gapfill_walker_context context = { .call.node = NULL, .count = 0 };
+
+	if (CMD_SELECT != parse->commandType || parse->groupClause == NIL)
+		return false;
+
+	if (!enable_partitionwise_aggregate)
+		return false;
+	/*
+	 * Only check for queries with gapfill call.
+	 */
+	gapfill_function_walker((Node *) parse->targetList, &context);
+
+	if (context.count == 0)
+		return false;
+
+	if (context.count > 1)
+		ereport(ERROR,
```

Comment:
Is this the right place to emit this error?

I mean, shouldn't this error have been raised prior to actually reaching the check for whether to push down the gapfill to data nodes? At least codecov seems to indicated this error is never reached in a test, so either:

1. This error cannot be reached (e.g., because an error is already raised earlier in the planner), so it should be replaced by an Assert.
2. A test is missing for this case